osu.Game/Rulesets/Edit/Checks/CheckBackground.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckBackground : Check
+    {
+        public override CheckMetadata Metadata { get; } = new CheckMetadata
+        (
+            category: CheckCategory.Resources,
+            description: "Missing background."
+        );
+
+        public override IEnumerable<IssueTemplate> PossibleTemplates => new[]
+        {
+            templateNoneSet,
+            templateDoesNotExist
+        };
+
+        private readonly IssueTemplate templateNoneSet = new IssueTemplate
+        (
+            type: IssueType.Problem,
+            unformattedMessage: "No background has been set."
+        );
+
+        private readonly IssueTemplate templateDoesNotExist = new IssueTemplate
+        (
+            type: IssueType.Problem,
+            unformattedMessage: "The background file \"{0}\" is does not exist."
+        );
```

Comment:
These should be constructed on use. Probably a good case to make nested classes for each of these. `PossibleTemplates` would just construct a copy of each class.

Reply:
Changed to nested classes, where `PossibleTemplates` receives copies of these constructed in the constructor of the check. This way the check can reuse the template to produce multiple issues, rather than creating a new instance for each.

This approach also helps with the other two suggestions.